packages/lib/CalendarManager.ts

```diff
@@ -151,10 +151,70 @@ const cleanIntegrationKeys = (
   return rest;
 };
 
+/**
+ * This function deduplicates credentials based on selected calendars
+ * It removes regular credentials for which corresponding delegation credentials exist which can be identified only associating them with selected calendars
+ *
+ * This is important to prevent unnecessary/duplicate calls to calendar APIs
+ */
+export const deduplicateCredentialsBasedOnSelectedCalendars = ({
```

Comment:
It is needed because a user can have a regular connection setup already and when we enable Delegation Credentials, there is another credential available for the same user, that basically causes double querying to Google Calendar